crates/precompile/src/secp256k1.rs

```diff
@@ -8,7 +8,31 @@ pub const ECRECOVER: PrecompileWithAddress = PrecompileWithAddress(
 
 pub use self::secp256k1::ecrecover;
 
-#[cfg(not(feature = "secp256k1"))]
+#[cfg(all(target_os = "zkvm", target_vendor = "succinct"))]
+#[allow(clippy::module_inception)]
+mod secp256k1 {
+    use crate::B256;
+    use revm_primitives::{alloy_primitives::B512, keccak256};
+
+    pub fn ecrecover(sig: &B512, recid: u8, msg: &B256) -> Result<B256, anyhow::Error> {
+        let mut sig_plus_recid = [0u8; 65];
+        sig_plus_recid[..64].copy_from_slice(&sig.0);
+        sig_plus_recid[64] = recid;
+        // TODO: Can we clean up the interface for this logic?
+        let recovered_key = sp1_precompiles::secp256k1::ecrecover(&sig_plus_recid, msg)?;
+
+        let mut hash = keccak256(&recovered_key[1..]);
+
+        // truncate to 20 bytes
+        hash[..12].fill(0);
+        Ok(hash)
+    }
+}
```

Comment:
@mattsse @Rjected we should show to Succinct how to do this precompile override with EVMBuilder, similar to 
* https://github.com/paradigmxyz/alphanet/blob/main/crates/node/src/evm.rs#L66-L82
* https://github.com/paradigmxyz/alphanet/blob/main/crates/node/src/evm.rs#L114
cc @ratankaliani